core/src/main/java/com/sopt/core/designsystem/component/timetable/NoostakEditableTimeTable.kt

```diff
@@ -0,0 +1,156 @@
+                backgroundColor = backgroundColor,
+                text = text,
+                onClick = {
+                    if (cellType == CellType.Data) {
+                        val cell = rowIndex to columnIndex
+                        if (selectedCells.contains(cell)) {
+                            selectedCells.remove(cell)
+                        } else {
+                            selectedCells.add(cell)
+                        }
+                        onSelectionChange(
+                            selectedCells.toTimeEntities(
+                                data.startTime,
+                                data.timeEntity
+                            )
+                        )
+                    }
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun getEditableBackgroundColor(
+    cellType: CellType,
+    rowIndex: Int,
+    columnIndex: Int,
+    isSelected: Boolean
+): Color {
+    return when (cellType) {
+        CellType.Blank, CellType.DateHeader, CellType.TimeHeader -> Color.Transparent
+        CellType.Data -> if (isSelected) NoostakTheme.colors.blue400 else Color.Transparent
+    }
+}
+
+@Composable
+fun NoostakEditableTimeTableBox(
+    index: Int,
+    days: Int,
+    timeSlots: Int,
+    backgroundColor: Color,
+    text: String,
+    onClick: () -> Unit
+) {
+    val shape = when (index) {
+        0 -> RoundedCornerShape(topStart = 10.dp)
+        days -> RoundedCornerShape(topEnd = 10.dp)
+        (days + 1) * timeSlots -> RoundedCornerShape(bottomStart = 10.dp)
+        (days + 1) * (timeSlots + 1) - 1 -> RoundedCornerShape(bottomEnd = 10.dp)
+        else -> RoundedCornerShape(0.dp)
+    }
+
+    Box(
+        modifier = Modifier
+            .border(
+                width = 0.5.dp,
+                color = NoostakTheme.colors.gray200,
+                shape = shape
+            )
+            .background(
+                color = backgroundColor,
+                shape = shape
+            )
+            .defaultMinSize(minWidth = 42.dp, minHeight = 36.dp)
+            .noRippleClickable { onClick() },
+        contentAlignment = Alignment.Center
+    ) {
+        Text(
+            modifier = Modifier.padding(horizontal = 5.dp, vertical = 3.dp),
+            text = text,
+            color = NoostakTheme.colors.gray600,
+            style = NoostakTheme.typography.c4Regular,
+            textAlign = TextAlign.Center,
+            maxLines = 2
+        )
+    }
+}
+
+fun List<Pair<Int, Int>>.toTimeEntities(
+    startTime: String,
+    timeEntityList: List<TimeEntity>
+): List<TimeEntity> {
+    val startHour = startTime.split(":")[0].toInt()
+    val groupedByColumn = this.groupBy { it.second }
+
+    return groupedByColumn.mapNotNull { (columnIndex, cells) ->
+        val date = timeEntityList.getOrNull(columnIndex - 1)?.date ?: return@mapNotNull null
+        val times = cells.map { cell ->
+            val hour = startHour + (cell.first - 1)
```

Comment:
P1: 요기 cell 말고 기존 rowIndex 사용하는게 더 이해하기 좋을 것 같슴다!

```suggestion
        val times = cells.map { (rowIndex, _) ->
            val hour = startHour + (rowIndex - 1)
```